robyn/router.py

```diff
@@ -69,49 +75,76 @@ def _format_response(self, res):
                 body=str(res).encode("utf-8"),
             )
         return response
-
+    #certain methods like add_route and other decorators
+    #return self._add_route(HttpMethod.GET, endpoint, handler, const)
     def add_route(
         self,
         route_type: HttpMethod,
         endpoint: str,
-        handler: Callable,
+        handler: Callable, #hello world function
         is_const: bool,
+        dependencies: Dict[str,any],
         exception_handler: Optional[Callable],
     ) -> Union[Callable, CoroutineType]:
         @wraps(handler)
         async def async_inner_handler(*args):
+            signatureObj = (inspect.signature(handler))
+            argsFromHandler = signatureObj.parameters.values() #holds all args from func args
+            specificDep = dependencies.get(endpoint, dependencies["all"])
+            depToPass = [] 
+            paramList = [a.name for a in argsFromHandler]
+            if any(key not in specificDep for key in paramList):
+                missing_deps = [key for key in paramList if key not in specificDep]
+                raise ValueError("Parameter not specified in dependencies dictionary:", missing_deps)
+            depToPass = [specificDep[key] for key in paramList if key in specificDep]
             try:
-                response = self._format_response(await handler(*args))
+                if depToPass:
+                    response = self._format_response(await handler( *depToPass))#next(iter(depToPass.values()))))
```

Comment:
do we need this?

Reply:
Do you mean the if/else statement? If so, then probably not. Could probably just be reduced to:
`try:`
`   response = self._format_response(await handler( *depToPass))`

but if you mean, "does it have to be *depToPass inside of await handler(...) instead of *args?", then yes. Unless you might have some suggestions for that.

Reply:
No, I was just talking about the comment 😅

Reply:
Ah ok, haha